docs/source/gateway.md

```diff
@@ -0,0 +1,118 @@
+states are accessed and hence which policies will need to be combined (including any individual state-based policies 
+as stored in the peer's copy of the ledger).  
+- This information is assembled into a `ChaincodeInterest` protobuf structure and passed to the discovery service in order
+to derive an endorsement plan specific for this transaction.
+- Using the endorsement plan, the gateway will connect to and request endorsement from other organizations' peers to satisfy
+the overall endorsement policy.  It will generally request endorsement from the peer with the highest block height for any
+given organization.
+
+If the transaction proposal contains transient data, then the gateway takes a more cautious approach.
+Transient data is often used to carry sensitive or personal information that must not persist in the ledger.
+A typical usage pattern involves sending the proposal to specific organizations which write the data to their private collections.
+The gateway will restrict the set of endorsing organizations to those that own the collections involved.
+If this restriction does not satisfy the endorsement policy, it will return an error to the client in preference
+to potentially leaking sensitive data.  In this situation, the client should [explicitly define which organizations should 
+endorse](#how-to-override-endorsement-policy-detection) the transaction.
+
+### How to override endorsement policy detection
+
+There may be situations where the client application needs to explicitly define which organizations should be targeted to
+evaluate or endorse a transaction proposal.  One such example is when sensitive data is being passed in the transient field,
+and the client wants to be *absolutely sure* which organizations will receive it.  In these situations the client application
+can specify the endorsing organizations. The gateway will select a peer from each of these (bypassing the mechanism described above)
+and build a transaction using these endorsements.  It's important to note that if this set of organizations does not satisfy
+the endorsement policy, then validation will fail later after ordering, and a failed transaction will be recorded on the ledger.
+
+### Error and retry handling
+
+#### Retry
+The gateway, using information from the discovery service, will make every effort to evaluate or submit a transaction in a network
+that could suffer from peers or ordering nodes being unavailable for any reason.  If a peer fails, and its organization is 
+running multiple peers, then another one will be selected.  If an organization fails to endorse entirely, then another
+set of organizations which also satisfies the endorsement policy will be targeted.  Only if there is no combination of available peers
+that satisfy the endorsement policy will the gateway stop retrying.  The gateway will not retry the same peer for 
+any given transaction proposal.
```

Comment:
I have several questions related to this, they don't all necessarily need to be documented, but I think some of them should be.

The gateway will really retry forever if there are many org-peer combinations that satisfy the endorsement policy?
Does the gateway keep the successful endorsements and just try to replace the failed endorsements?
Or will it gather a whole new set of endorsements potentially using different organizations?
What if you get a chaincode success from one org and a chaincode failure from another org (rather than the peer being unavailable)?

Reply:
> The gateway will really retry forever if there are many org-peer combinations that satisfy the endorsement policy?

It will try all combinations, but...

> Does the gateway keep the successful endorsements and just try to replace the failed endorsements at the same org(s)?

Yes

> Or will it gather a whole new set of endorsements potentially using different organizations?

No

> What if you get a chaincode success from one org and a chaincode failure from another org (rather than the peer being unavailable)?

It will keep the success, but retry failures, even if it means trying other layouts

Often a 'group' (org) will appear in multiple layouts, so when a successful endorsement is obtained for that group, it could partially satisfy multiple layouts at once.  For every successful endorsement received, it will check the status of all layouts in a plan - it only needs one of them to be satisfied, then it uses that.

Reply:
Not sure how much of that to document.  This whole section is "good to know", but not essential knowledge for using the gateway.  While it's good to demystify the 'magic', I don't want to tie us into a particular implementation that we can change in the future.